src/input/mod.rs

```diff
@@ -1815,22 +1815,30 @@ impl State {
                         }
                     }
                 }
-
                 return;
             } else {
                 self.niri.horizontal_finger_scroll_tracker.reset();
                 self.niri.vertical_finger_scroll_tracker.reset();
             }
         }
 
-        let horizontal_amount = horizontal_amount.unwrap_or_else(|| {
-            // Winit backend, discrete scrolling.
-            horizontal_amount_v120.unwrap_or(0.0) / 120. * 15.
-        });
-        let vertical_amount = vertical_amount.unwrap_or_else(|| {
-            // Winit backend, discrete scrolling.
-            vertical_amount_v120.unwrap_or(0.0) / 120. * 15.
-        });
+        let scroll_factor = match source {
+            AxisSource::Wheel => self.niri.config.borrow().input.mouse.scroll_factor,
+            AxisSource::Finger => self.niri.config.borrow().input.touchpad.scroll_factor,
+            _ => 1.0,
+        };
+
+        let horizontal_amount = scroll_factor
+            * horizontal_amount
+                // Winit backend, discrete scrolling.
+                .or(horizontal_amount_v120.map(|a| a / 120. * 15.))
+                .unwrap_or(0.0);
+
+        let vertical_amount = scroll_factor
```

Comment:
Could you move `scroll_factor *` in the end so it's `vertical_amount.unwrap_or_else(...) * scroll_factor;`? I think that should preserve the previous formatting.